CHANGES.current

```diff
@@ -7,6 +7,22 @@ the issue number to the end of the URL: https://github.com/swig/swig/issues/
 Version 4.3.0 (in progress)
 ===========================
 
+2024-02-29: erezgeva
+            #2609 Fix Java typemap (char *STRING, size_t LENGTH) to use string type.
+            Add STRING-LENGTH typemap to most languages.
+            All languages uses string type for STRING-LENGTH typemap.
+            Add char_binary test to most languages and
+             director_binary test to languages supporting director class.
+            Add new typemap BYTES-LENGTH to use with cdata.
+            Add BYTES-LENGTH typemap to most languages.
+            Rebase cdata typemap to improve coherent among different languages.
+            Add li_cdata test to most languages.
+            Languages with static type use byte array for cdata.
+            While dynamic type languages retain string.
+            incompatibilities:
+            * STRING-LENGTH: Java uses String
+            * cdata: Go uses byte array and int64 for size
+
```

Comment:
This style of docs should be more exact. 'STRING-LENGTH' style is not something used anywhere in SWIG. It needs to be (I think): `(const char *STRING, int LENGTH)` multi-argument typemaps, or maybe? `(const char *STRING, size_t LENGTH)`.

For the incompatibilities, clear, full examples of what users need to change is required.

Reply:
and what is needed to restore the old functionality if a user needs it.

Reply:
Can you look on my update?
What do you think?

And what about using `Cdata(m, int64(7))` in go?
Do you think we need to use `int` for go?